clinica/iotools/converters/nifd_to_bids/nifd_utils.py

```diff
@@ -1,309 +1,327 @@
-    Args:
-        path: Path to a Dicom file
-        equivalences: dictionary, equivalences['medical_image_name'] = (Descriptor_instance, modalityLabel)
+    # Keep relevant columns and rename them.
+    dataframe = (
+        dataframe[["dx", "site", "education", "race", "cdr_box_score", "mmse_tot"]]
+        .rename(columns={"dx": "diagnosis", "cdr_box_score": "cdr", "mmse_tot": "mmse"})
+        .astype(
+            dtype={
+                "diagnosis": CategoricalDtype(
+                    ["BV", "CON", "L_SD", "PATIENT (OTHER)", "PNFA", "SV"]
+                ),
+                "site": CategoricalDtype(["UCSF", "MAYO", "MGH"]),
+                "education": "int",
+                "race": "int",
+                "cdr": "float",
+                "mmse": "float",
+            }
+        )
+        .replace({"education": {99: NA}, "race": {50: NA, 99: NA}})
+    )
+
+    # Keep positive MMSE values only.
+    dataframe.mmse = dataframe.mmse.mask(dataframe.mmse < 0)
 
-    Returns:
-        Medical name
-    """
-    path = break_path(path)
-    for i in path:
-        if i in equivalences:
-            return i
+    return dataframe
 
 
-def print_patients_source_files(dict):
-    """Pretty printer for a nested dictionary such as patients_source_files.
+def find_collection_data(imaging_data_directory: PathLike) -> Optional[DataFrame]:
+    from pathlib import Path
 
-    Args:
-        dict: Data structure of the form dict['subject_ID'] = [paths/to/dcm]
-    """
-    from clinica.utils.stream import cprint
+    from pandas import read_csv
 
-    for key in dict:
-        cprint(key)
-        for elt in dict[key]:
-            cprint("\t" + str(elt))
+    dataframe = None
+    for path in Path(imaging_data_directory).rglob("*.csv"):
+        try:
+            dataframe = (
+                read_csv(
+                    path,
+                    index_col="Image Data ID",
+                    parse_dates=["Acq Date"],
+                )
+                .rename(columns=lambda x: x.lower().replace(" ", "_"))
+                .rename_axis(index=lambda x: x.lower().replace(" ", "_"))
+                .convert_dtypes()
+                .sort_index()
+            )
+        except (ValueError, IndexError):
+            continue
 
+    return dataframe
 
-def print_orderedBIDS(dict):
-    """Pretty printer for a nested dictionary such as ordered_bids, final_bids.
 
-    Args:
-        dict: Data structure of the form dict['session_id']['dataType']['Priority']['Final_name'] = [paths/to/dcm]
-    """
-    from clinica.utils.stream import cprint
+def find_imaging_data(imaging_data_directory: PathLike) -> Iterable[Tuple[str, str]]:
+    import re
+    from pathlib import Path
+
+    pattern = re.compile(
+        r"NIFD_(?:\d_S_\d{4})_(?:\w+)_(?:\d{4}_\d{2}_\d{2})_(?:\d+)_(?:S\d+)_(I\d+)$"
+    )
+
+    for path in Path(imaging_data_directory).rglob("NIFD*"):
+        if path.is_dir():
+            # path = path.relative_to(imaging_data_directory)
```

Comment:
Can this line be removed? Or if you consider important to keep it, can you add a comment?